website/docs/contributing/ADRs/overarching/separation-request-response-schemas.md

```diff
@@ -0,0 +1,41 @@
+
+The separation of request and response schemas will provide the following benefits:
+
+1. **Enhanced clarity and correctness**: With dedicated schemas for requests and responses, we can define the precise structure and constraints for each interaction. This will help prevent situations where the schemas are overly permissive or restrictive, improving the accuracy and clarity of the API documentation.
+
+2. **Improved maintainability**: By avoiding the reuse of schemas between requests and responses, we can modify and update them independently. This decoupling of schemas will simplify maintenance efforts and minimize the risk of unintended side effects caused by changes in one context affecting the other.
+
+3. **Flexibility for future enhancements**: Separating request and response schemas lays the foundation for introducing additional validation or transformation logic specific to each type of interaction. This modularity will enable us to incorporate future enhancements, such as custom validation rules or middleware, with ease.
+
+### Concerns
+
+While this decision brings several benefits, we acknowledge the following concerns that may arise from the separation of request and response schemas:
+
+1. **Increased schema maintenance**: By having separate schemas for requests and responses, there will be a need to maintain and update two sets of schemas instead of a single shared schema. This could potentially increase the maintenance overhead and introduce the possibility of inconsistencies between the two schemas.
+
+2. **Data duplication and redundancy**: With the separation of schemas, there might be instances where certain data fields or structures are duplicated between the request and response schemas. This redundancy could lead to code duplication and increase the risk of inconsistencies if changes are not carefully synchronized between the two schemas.
+
+3. **Complexity of migration**: Migrating existing API implementations to adopt the separated schemas might introduce complexities and challenges. Ensuring a smooth transition without disrupting ongoing development or existing integrations requires careful planning and coordination.
+
+4. **Developer adoption and understanding**: Developers who are accustomed to the previous unified schema approach may initially find it challenging to adapt to the new separation of schemas. This could potentially lead to confusion and frustration if the new approach is not clearly communicated and explained.
+
+## Conclusion
+
+By implementing the separation of request and response schemas, we aim to improve the robustness and maintainability of our API documentation. This decision will empower developers to build more reliable integrations by providing clearer guidelines for both request and response data structures.
```

Comment:
I think it's worth mentioning that this also has value to us, internally. This should allow us to write more robust code and do less massaging of incoming responses into the correct shapes. That in turn reduces the likelihood of bugs and makes the code significantly easier to reason about and work with

Reply:
Sorry, I think we're conflating terms here. Incoming responses doesn't make sense to me. We are talking incoming requests and outgoing responses no?

The point still stands that preparing responses will be easier, and requests will accept anything, but have a predefined set of fields we'll actually need to look at, any superfluous fields can just be dropped.

Reply:
Yuh, thanks Chris! I meant to say incoming requests, edited my original comment to reflect that, leaving this here so that your comment makes sense to people who come after

Reply:
Thanks, tried addressing this here: https://github.com/Unleash/unleash/pull/3869/commits/70148a3a3ea2d6804f82e1ee5a13d8125ea5299d